sphinx/domains/c/_symbol.py

```diff
@@ -101,18 +101,43 @@ def __init__(
         self.isRedeclaration = False
         self._assert_invariants()
 
-        # Remember to modify Symbol.remove if modifications to the parent change.
-        self._children: list[Symbol] = []
-        self._anonChildren: list[Symbol] = []
-        # note: _children includes _anonChildren
+        # These properties store the same children for different access
+        # patterns. Symbol._add_child and Symbol._remove_child should be
+        # used for modifying them.
+        self._childrenByName: dict[str, Symbol] = {}
+        self._childrenByDocname: dict[str, dict[str, Symbol]] = {}
+        self._anonChildren: set[Symbol] = set()
```

Comment:
Just wondering, but is it possible that the resolution could have different results if the anonymous children are not ordered in the order they were discovered?